subsys/usb/device_next/usbd_core.c

```diff
@@ -77,6 +77,9 @@ static void usbd_class_bcast_event(struct usbd_context *const uds_ctx,
 
 	SYS_SLIST_FOR_EACH_CONTAINER(&cfg_nd->class_list, c_nd, node) {
 		switch (event->type) {
+		case UDC_EVT_VBUS_REMOVED:
+			usbd_class_disable(c_nd->c_data);
```

Comment:
This breaks the internal state if device was configured. I think it would be much better to issue missing reset on VBUS removal in udc driver as a vendor quirk.

Reply:
I reworked it to align with your suggestion. It works but there's one bug related to disconnecting the cable while still sending reports. I will have a look at it.

Reply:
Converting to draft for now.